src/poetry/publishing/uploader.py

```diff
@@ -61,9 +61,21 @@ def user_agent(self) -> str:
         agent: str = user_agent("poetry", __version__)
         return agent
 
+    @property
+    def dist_dir(self) -> Path:
+        io_dist_dir = "dist"
+        if "dist-dir" in self._io.input.options:  # Option comes with publish command.
```

Comment:
I think it would be better to put dist dir as an optional init argument for Uploader.

Reply:
Yeah, I suppose we want to try to access command line options only in commands (via `self.option()`) and pass them explicity to classes that need them.

Reply:
What about now?